test/e2e/lib/broker.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lib
+
+import (
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/knative-gcp/test/e2e/lib/metrics"
+	pkgmetrics "knative.dev/pkg/metrics"
+)
+
+func AssertBrokerMetrics(t *testing.T, client *Client) {
+	sleepTime := 4 * time.Minute
```

Comment:
This is a long time! Where did this value come from?

Reply:
Observations that data would show up about 2 minutes after the request was processed and wanting a buffer to make sure the tests won't be flaky. I don't like it, but it makes the test pass.

Reply:
Thanks for figuring out how to make this pass!